### ISSUE TITLE
HTTP Proxy Support

### ISSUE THREAD

**Issue by zicklag** [theduke/crates_io_api#18]
Related to this issue: https://github.com/crev-dev/cargo-crev/issues/293.

---

It appears that `crates_io_api` doesn't work behind an HTTP proxy. It would be great if it would read the `http_proxy`, `https_proxy`, `HTTPS_PROXY`, `HTTP_PROXY`, `no_proxy`, and `NO_PROXY` environment variables.

**Comment by theduke** [comment]
I feel like reading such env vars is out of scope for a library. This should be done by cargo-crev.

Will need to add a config setting, but that's an easy fix.

**Comment by zicklag** [comment]
I've actually got a better idea. All you have to do is upgrade `reqwest` to `0.10` and it will automatically read the proxy settings so you don't have to do anything. I'm working on a PR. `reqwest` 0.10 requires futures 0.3, but there is a compatibility layer so it shouldn't be difficult to integrate and will prevent having to switch this whole crate to use futures 0.3.

**Comment by theduke** [comment]
Even better. See #17 .

**Comment by zicklag** [comment]
I just got it working. PR incoming. :)

For now it is using the futures compatibility layer, but once #17 is ready it will be trivial to take out futures 0.1 and the compat layer and just use futures 0.3.

**Comment by theduke** [comment]
Published as `0.6`.
